### ISSUE TITLE
Expose the store as a Ember.Service

### ISSUE THREAD

**Issue by martndemus** [emberjs/data#2818]
It would be very neat to be able to do this:
```javascript
App.FooObject = Ember.Object.extend({
  store: Ember.inject.service(),
  getSomething: function() {
    return this.get('store').find('something');
  }
});
```

**Comment by Kuzirashi** [comment]
+1 on that. It would be nicer than using `targetObject.store` in components.

**Comment by rwjblue** [comment]
Yes, I would like to be able to do this also.  I've discussed this previously with @tomdale, and I believe he is :+1: on this conceptually.  I am not sure what will be required to make it work properly though (haven't spent time looking into it much).

**Comment by quaertym** [comment]
:+1:  This would be great if added.

**Comment by martndemus** [comment]
I just played around get it to work, but this works for me:

```javascript
// initializers/store-as-a-service.js
export default {
  name: 'store-as-a-service',
  after: 'ember-data',
  initializer: function(container, application) {
    var store = container.lookup('store:main')
    container.register('service:store', store, { instantiate: false });
  }
}

**Comment by bmac** [comment]
@martndemus Would you like to submit a pr adding similar logic to the end of this file?
https://github.com/emberjs/data/blob/master/packages/ember-data/lib/initializers/store.js#L36

**Comment by martndemus** [comment]
Done, see #2820

**Comment by bmac** [comment]
Fixed by #2820.
